src/controllers/DefaultController.php

```diff
@@ -87,7 +87,7 @@ public function actionIndex(string $pattern): Response
             $cache = ArrayHelper::remove($config, 'cache', false);
 
             if ($cache) {
-                $cacheKey = 'elementapi:'.$request->getPathInfo().':'.$request->getQueryStringWithoutPath();
+                $cacheKey = 'elementapi:'.$request->getPathInfo().':'.$request->getQueryStringWithoutPath().':'.$site;
```

Comment:
Let's move the `$siteId` part of the key to after `elementapi:` and before the path info so it's going from a wider scope to a narrower one.